### ISSUE TITLE
detection.stops detecting stops below defined minutes_for_a_stop

### ISSUE THREAD

**Issue by sianteesdale** [scikit-mobility/scikit-mobility#138]
Hello! I am currently using your package to detect stops with my GPS mobile phone data (which is 19.9 million rows long). I followed the pre-processing instructions by first converting my DataFrame to TrajDataFrame:

```
df_skmob = skmob.TrajDataFrame(df, latitude='LATITUDE', longitude='LONGITUDE', user_id='TRAJECTORY_ID',
                                 datetime='Timestamp')
```

Then filtering out points over 130km/h:
```
df_skmob = filtering.filter(df_skmob, max_speed_kmh = 130.)
```
Before finally doing the stop detection function like this:
```
stop_detect = detection.stops(df_skmob, stop_radius_factor=0.5, 
                                minutes_for_a_stop=10.0, spatial_radius_km=0.05, 
                                leaving_time=True, min_speed_kmh = 3)
```

which detects 299,508 stops within my dataset. However, upon investigating the length of duration of these stops, I've discovered that the function has detected 6,221 stops that have a stop duration of less than 10 minutes, which I defined in the above function. 

I am fine with filtering out the stops that are less than 10 minutes but thought it was worth highlighting in case it was a bug with the stop detection algorithm. Many thanks in advance!

**Comment by FilippoSimini** [comment]
Could you please provide a minimal example (i.e. a dataset with a dozen rows) to reproduce your issue? Thanks.

**Comment by sianteesdale** [comment]
Hi Filippo, unfortunately I am unable to share my personal data as it would breach my data contract with the data providers. However, I have just tested it on the Microsoft Geolife GPS dataset and I am getting the same error there.

Here's the data: [geolife_data.zip](https://github.com/scikit-mobility/scikit-mobility/files/5667132/geolife_data.zip)

Once I've used the detection.stops algorithm, I have to convert the leaving_datetime column to datetime as sometimes it is in object format:

```
stop_detect = pd.DataFrame(stop_detect)

stop_detect['leaving_datetime'] = pd.to_datetime(stop_detect['leaving_datetime'], 
                                                 format = '%Y-%m-%d %H:%M:%S')
```
and then work out the stop duration and filter the durations below 10 minutes:
```
stop_detect['stop_dur'] = (stop_detect['leaving_datetime'] - stop_detect['datetime']).astype('timedelta64[m]')

stop_detect.loc[stop_detect.stop_dur < 10]
```

Hope this helps. Thanks for replying!
